sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -2516,4 +2516,26 @@ class HiveDDLSuite
       }
     }
   }
+
+  test("SPARK-28313: Spark sql null type incompatible with hive void type") {
+    val expectedMsg = "DataType void is not supported"
+    withTable("t") {
+      val e = intercept[AnalysisException] {
+        sql("create table t (a void)")
+      }.getMessage
+      assert(e.contains(expectedMsg))
+    }
+
+    withTable("t") {
+      sql("CREATE TABLE t AS SELECT NULL AS col ")
```

Comment:
what's the behavior before your patch?

Reply:
Throw a SparkException 'Cannot recognize hive type string: NULL'